pkg/services/blockfetcher/blockfetcher.go

```diff
@@ -0,0 +1,203 @@
+	cid "github.com/nspcc-dev/neofs-sdk-go/container/id"
+	"github.com/nspcc-dev/neofs-sdk-go/object"
+	oid "github.com/nspcc-dev/neofs-sdk-go/object/id"
+	"go.uber.org/zap"
+)
+
+type Service struct {
+	chain       *core.Blockchain
+	log         *zap.Logger
+	client      *client.Client
+	db          storage.Store
+	quit        chan bool
+	containerID cid.ID
+	Timeout     time.Duration
+	Nodes       []string
+	dumpDir     string
+}
+
+// ConfigObject represents the configuration object in NeoFS.
+type ConfigObject struct {
+	HashOID   string   `json:"hash_oid"`
+	BlockOIDs []string `json:"block_oid"`
+	Height    uint32   `json:"height"`
+	Timestamp int64    `json:"timestamp"`
+	Step      uint32   `json:"step"`
+}
+
+// Config is the configuration of the BlockFetcherService.
+type Config struct {
+	ContainerID string
+	Nodes       []string
+	Timeout     time.Duration
+	DumpDir     string
+}
+
+// New creates a new BlockFetcherService.
+func New(chain *core.Blockchain, store storage.Store, cfg Config, logger *zap.Logger) *Service {
+	neofsClient, err := client.New(client.PrmInit{})
+	if err != nil {
+		logger.Error("Failed to create NeoFS client", zap.Error(err))
+		return nil
+	}
+	var containerID cid.ID
+	err = containerID.DecodeString(cfg.ContainerID)
+	if err != nil {
+		logger.Error("Failed to decode container ID", zap.Error(err))
+		return nil
+	}
+	return &Service{
+		chain:       chain,
+		log:         logger,
+		client:      neofsClient,
+		db:          store,
+		quit:        make(chan bool),
+		dumpDir:     cfg.DumpDir,
+		containerID: containerID,
+		Nodes:       cfg.Nodes,
+		Timeout:     cfg.Timeout,
+	}
+}
+
+// Name implements the core.Service interface.
+func (bfs *Service) Name() string {
+	return "BlockFetcherService"
+}
+
+// Start implements the core.Service interface.
+func (bfs *Service) Start() {
+	bfs.log.Info("Starting Block Fetcher Service")
+	err := bfs.fetchData()
+	if err != nil {
+		close(bfs.quit)
+		return
+	}
+}
+
+// Shutdown implements the core.Service interface.
+func (bfs *Service) Shutdown() {
+	bfs.log.Info("Shutting down Block Fetcher Service")
+	close(bfs.quit)
+}
+
+func (bfs *Service) fetchData() error {
+	for {
+		select {
+		case <-bfs.quit:
+			return nil
+		default:
+			prm := client.PrmObjectSearch{}
+			filters := object.NewSearchFilters()
+			filters.AddFilter("type", "config", object.MatchStringEqual)
+			prm.SetFilters(filters)
+
+			configOid, err := bfs.search(prm)
+			if err != nil {
+				bfs.log.Error("Failed to fetch object IDs", zap.Error(err))
+				return err
+			}
+			cfg, err := bfs.get(configOid[0].String())
+			var configObj ConfigObject
+			err = json.Unmarshal(cfg, &configObj)
+			if err != nil {
+				bfs.log.Error("Failed to unmarshal configuration data", zap.Error(err))
+				return err
+			}
+			_, err = bfs.get(configObj.HashOID)
+			if err != nil {
+				bfs.log.Error("Failed to fetch hash", zap.Error(err))
+				return err
+			}
+			for _, blockOID := range configObj.BlockOIDs {
```

Comment:
Do you retrieve all hashes in a single request? What is the scheme of data stored in NeoFS?